src/float.rkt

```diff
@@ -62,6 +62,20 @@
     [_ (hash 'type (~a (representation-name repr))
              'ordinal (~a ((representation-repr->ordinal repr) x)))]))
 
+(define (json->value x repr)
+  (match x
+    [(? real?) x]
+    [(? hash?) 
+      (match (hash-ref x 'type) 
+        ["real"
+          (match (hash-ref x 'value)
+            ["-inf" -inf.0]
+            ["+inf" +inf.0]
+            ["NaN" +nan.0]
+            [_ +nan.0])]
```

Comment:
I would drop this case—probably better to crash on unknown input than do something strange.
```suggestion
            ["NaN" +nan.0])]
```